assets/prototype/domain/eventEditor/context/DateTimeProvider/useDateEditorId.ts

```diff
@@ -2,13 +2,13 @@
  * Internal dependencies
  */
 import { EntityId } from '../../data/types';
+import { EditorId } from '../../../../application/ui/components/layout/editor-modal/useEditorModalState/types';
 
 /**
  * Function that returns editor id for a given event date
  */
-const useDateEditorId = (dateId: EntityId, prefix: string = ''): string => {
-	const editorId: string = `event-date-editor-${dateId}`;
-	return prefix !== '' ? `${prefix}-${editorId}` : editorId;
+const useDateEditorId = (prefix: string = 'datetime-editor', datetimeId: EntityId): EditorId => {
+	return prefix !== '' ? `${prefix}-${datetimeId}` : `datetime-editor-${datetimeId}`;
```

Comment:
Just realized that since we have the default value for `prefix`,  the ``datetime-editor-${datetimeId}`` result never will occur :smiley:

Reply:
i thought that if someone passes an empty string it would negate the default, so I added it back in as a fallback.
that was my thought process... maybe that's not accurate and it's not necessary... haven't confirmed with tests yet 😆

Reply:
In continuation of your previous suggestion to have a more strict type system - I think that if we'd somehow enforce non-empty string type or a string utility helper.
That could make life easier :)
***
out of the scope of current PR :point_up:

Reply:
>  if we'd somehow enforce non-empty string type

can you do that when there is a default?

ie: if the type is set to string, then `''` is valid

Reply:
> can you do that when there is a default?
yeah, with a default value, that isn't possible
my comment was a general hope that TS would help to prevent some boilerplate code, like this(maybe):
```
if(variable !== undefined) {
  return doThis();
}

if(isString(variable) && isEmpty(variable)) {
  return doThat();  
}

if(isString(variable) && isNotEmpty(variable)) {
  return doWhatever();  
}
```
just thoughts out loud, not smth related to `useDateEditorId`